lib/mapObjects/CObjectHandler.cpp

```diff
@@ -215,6 +216,8 @@ void CGObjectInstance::initObj(CRandomGenerator & rand)
 		blockVisit = true;
 		break;
 	}
+
+	VLC->objtypeh->getHandlerFor(ID, subID)->setAmbient(this);
```

Comment:
Yeah I find out that doing that just after initObj call is way better since we initialize objects just in few places. Though there is problem that for whatever reason "CObjectClassesHandler::getHandlerFor" can't find handler for some objects and it's fail on runtime error.